apps/docs/content/components/FormControl.mdx

```diff
@@ -229,45 +232,24 @@ Try changing the input value to to `monalisa` to show the `success` state.
 
 ```javascript live noinline
 const App = () => {
-  const [value, setValue] = React.useState()
-  const [validationState, setValidationState] = React.useState()
-
-  React.useEffect(() => {
-    const defaultHandle = 'mona lisa'
-    setValue(defaultHandle)
-    validate(defaultHandle)
-  }, [])
-
-  const validate = (inputValue) => {
-    if (/\s/g.test(inputValue)) {
-      setValidationState('error')
-    } else {
-      setValidationState('success')
-    }
-  }
+  const [value, setValue] = React.useState('mona lisa')
 
   const handleChange = (event) => {
-    event.preventDefault()
-    if (!event.target.value) {
-      setValue(undefined)
-      setValidationState(undefined)
-      return
-    }
     setValue(event.target.value)
-    validate(event.target.value)
   }
 
+  const isInvalid = /\s/g.test(value)
```

Comment:
FYI we've lost the animation here, presumably because we're not forcing a re-render? Out of interest, was this intentional or just a side-effect of cleaning this example up?

Previous (current) production with animation:

https://github.com/user-attachments/assets/1595b8e1-e1db-456a-808a-19e94db6b736





Now with this regex approach:

https://github.com/user-attachments/assets/fb58a941-6ee7-4679-a275-adb2be8cc0a0